### DISCUSSION TITLE
Cannot implicity convert type 'void' to 'Stride.Core.Mathematics.Vector3'

### DISCUSSION THREAD

**Discussion by ghost** [stride3d/stride#1374] [Q&A] [answered]
i have a movement script, and i want to move forward in the direction i'm facing, i have this bit of code which i will pass to the character.set velocity in order to move. i'm not sure what to do to fix this error though.
![aaaaaaa](https://user-images.githubusercontent.com/89107810/159771448-fb9f375b-218f-465e-9e00-f0dab464e817.png)

**Comment by Eideren** [comment, marked as answer]
This is not an issue specific to stride, did you take a look on google, there are loads of resources online for it.
That function returns `void`, aka nothing. You're trying to assign `nothing` to `test`, you can't assign `nothing` to `something`.
Rotate rotates the vector you pass in since it's passed as `ref`.
